docs/developer-guide/running-substreams.md

```diff
@@ -77,3 +77,36 @@ map_transfers: message "eth.erc721.v1.Transfers": {
 ```
 
 The example output shown above contains data for different transfers from data in the blockchain. These transfers can also be verified on [Etherscan](https://etherscan.io/tx/0xcfb197f62ec5c7f0e71a11ec0c4a0e394a3aa41db5386e85526f86c84b3f2796).
+
+## Development & Production Mode
+
+### Development Mode
+
+Production and development modes impact the execution of Substreams. Key aspects of execution include:
+
+- The time required to reach the first byte.
+- The speed that large ranges get executed.
+- The module logs and outputs that are sent back to the client.
+
+Differences between production and development modes include:
+
+- Forward execution is enabled in production mode and disabled in development mode.
+- The time required to reach the first byte in development mode is faster than in production mode.
+- In development mode, the client will receive all of the executed module's logs. In production mode the client will only receive the output module's logs.
+- In development mode, it's possible to request specific store snapshots in the execution tree.
+
+Examples: (given the dependencies: `[block] --> [map_pools] --> [store_pools] --> [map_transfers])`
```

Comment:
This would require a quick sentence to introduce the examples, the transition feels really hard here. It seems also the bullet list below is all for development mode but no where the reader can know that.

Reply:
Subtitle/header added for this section. The changes will be available in the next commit/push.

Reply:
> Examples: (given the dependencies: `[block] --> [map_pools] --> [store_pools] --> [map_transfers])`

Still feels like it should be improved. I think it should be turned into a proper sentence or 2.